### DISCUSSION TITLE
Sort of easy but not

### DISCUSSION THREAD

**Discussion by tacklers** [drivardxrm/Dataverse-CustomApis#3] [General]
Hi David, I appreciate the work you have put into this. I listened to the XRM podcast twice, not easy as per usual people who know what they're talking about ramble off onto advanced matters such as privileges which for a beginner are really not important. A bit like the glitch in the gui when creating that they didn't appear . They do now in the new solution viewer.

Anyway. A good selection of API's and part of the purpose was to educate. Some fine features but some simple simple CRUD aps would have been nice. I can't see 'remove Diacritics' being up my street, to thos who it might appeal to theye probably way ahead in terms of competence anyway. 

I now have to work out how I can separate out the eight apps in one solution so I can build a tiny simple Crud op as a beginner.

Thanks. Richard UK

**Comment by drivardxrm** [comment]
Hi @tacklers , thanks for your interest in the project.

I don't really understand what you mean by simple CRUD apps in the context of this library/project , can you elaborate a bit more?
If you have ideas for any API that you think are missing please submit the idea, if it is relevant I will add them when I have the time.

As for your comment on separation, I don't get it.
This project is sort of a toolset that you can install on your environment. You are not forced to use what you dont wan't to use.

If you really want to install only what you need, just install the unmamaged solution from the latest release and delete the custom API that you dont need. and export the 'trimmed' solution in your environments

**Comment by tacklers** [comment]
Hi David,

I appreciate the feedback. It’s all a matter of expectations I guess.

I imagined your motivation was to help reveal how the Custom API’s could be exploited by those who don’t know. The same people who maybe aren’t pro-devs all the time in this space. I recall mention that lo-code devs could call them from automate etc.

I have searched high and low for a simple custom API that does a simple CRUD op, say create  record and return an ID. You don’t have one. I also don’t have the skills to be able to confidently unpick your big solution and split it up with only the references I need.

Yesterday I did find a ‘simple’ API with code to create a contact. Happy days I thought as the authors intention was to share knowledge to those less fortunate than himself. It started off okay then on the build I had missing references. I eventually find that I would need to load XrmExtension via XrmContext, to download a nuget package to install the service to do entity mapping in the early bound context etc. Sadly just before I lost the will to live I realised I didn’t have the bandwidth.

I really do appreciate how you guys have the talent and will to help educate but what might seem to you is understandable for newbies is off target. Shame.

Keep up the good work. Cheers, richard

**Comment by drivardxrm** [comment]
Concerning this comment

"I have searched high and low for a simple custom API that does a simple CRUD op, say create  record and return an ID. You don’t have one. I also don’t have the skills to be able to confidently unpick your big solution and split it up with only the references I need."

Not only that I don't **have** one... I don't **need** one.

Why would I reinvent the wheel, there is already the Dataverse builtin API that is meant for record creation. 
https://docs.microsoft.com/en-us/powerapps/developer/data-platform/webapi/overview

In Power Automate, just use the Dataverse connector its made for that
https://docs.microsoft.com/en-us/connectors/commondataserviceforapps/

**Comment by tacklers** [comment]
Hi David,

I hear where you’re coming from.

Sadly The CDSWebApiService example is just a console write line. No CRUD.
Very sadly we followed the documentation to the letter last week and incorporated a write record controller. It ran, obtained a token but failed to return a response as it hung on deleting the token.
Four clever C#/.NET guys and me looked it over then gave up after we repeated the exercise on a different machine and VS2019. Same hang.
Later in the day I found an MVP had the same experience and turned to the Organisation service approach as we have done.

I find it hard to believe that CDSWebApiService is flawed but we didn’t have the time or knowledge to suss it.

We’re now on the Custom API path and I’m still stuck on Plugin CRUD ops from an API but also wonder what I have to configure on the Azure front end to talk to the Dataverse Custom API.

There’s a lot of building blocks out there but very few pictures to help join it all up.

Cheers, richard

**Comment by drivardxrm** [comment]
I'm sorry to ear that but by reading your comment , it seems that you are mixing up a lot of basic Dataverse concepts.

Creating a basic CRUD app that targets the Dataverse is something somehow trivial for any experienced Dataverse developpers.

**Comment by tacklers** [comment]
Agreed.

We’ve just got skewed by looking to get a .NET app running on an aws instance to pass a datapacket via an API into Dataverse via an Azure front end located on a non-public VPN.

As agile lo-code-pro-devs we exploit the platform but avoid stuff like C# code whenever we can.

This middle band of people need all the help we can get!

Hence the lookout for guys like you clever enough to do it all the time to show us the ropes.

All looks good on the MVP ladder!
